### DISCUSSION TITLE
Redirect traffic to different services according to http method used

### DISCUSSION THREAD

**Discussion by alexBonadei** [mia-platform/community#137] [Q&A] [answered]
Hi,

I need to manage traffic according to the method used in the request. Say for example I want that all the GET /books requests go to the crud-service and the POST / books requests go the book-service which contains custom logic.

The endpoint /books exposes the crud-service. This endpoint rewrites the path accordingly to the collection name in Mongo, say for example: rewrite-base-path = /book_entities. So by default GET /books goes to the crud-service.

I tried to configure this map in the `api-gateway/maps-proxyName.before.map` extension file to manage the POST requests:
```
"~^(secreted|unsecreted)-1-POST-/books" "book-service";
``` 

When I try to call POST /books, the request is correctly redirected to the book-service, but the path is rewritten. The request path received by the book-service is then /book_entities. 
I would expect that the book-service will receive /books path.

Am I missing something?
Thanks in advance

**Comment by fredmaggiowski** [comment]
Hi!

If you need a different rewrite you need to also define the `api-gateway/maps-proxyUrl.before.map`

**Reply by alexBonadei** [reply]
Yes I was missing that! Thanks a lot!

**Comment by davidebianchi** [comment, marked as answer]
You need to rewrite also the `api-gateway/maps-proxyUrl.before.map`. Something like:

```
"~^POST-/books(?<path>[/\?].*|$)$" "/books$path";
```

If you expose it under the book-service service under the `/books` path

**Reply by alexBonadei** [reply]
Yes, that's definitely  what I was missing. Thanks a lot!
